tests/regression/ba-issues/issues/issue-4644/test.wat

```diff
@@ -0,0 +1,24 @@
+;; 定义数组类型 - i32数组
+(type $i32_array (array (mut i32)))
+
+;; main函数
+(func $main
+  (local $arr (ref $i32_array))
+
+  ;; 创建一个大小为10的i32数组，初始值为0
+  (array.new $i32_array
+    (i32.const 0)    ;; 初始值
+    (i32.const 10))  ;; 数组大小
+  (local.set $arr)
+
+  ;; 使用 array.fill 填充数组
+  (array.fill $i32_array
+    (local.get $arr)  ;; 数组引用
+    (i32.const 0)     ;; 起始索引
+    (i32.const 42)    ;; 填充值
+    (i32.const 10))   ;; 填充长度 - 填满整个数组
+)
+
```

Comment:
👍 love it

And if it's convenient for you, please write comments in English for better readability by others.